index.js

```diff
@@ -26,7 +26,7 @@ function _fillMemDuplex(memDuplexes, index, dataRetrievalFn, response, logger){
         (err, readable) => {
         if(err){
             logger.error('failed to get full object');
-            return response.end();
+            throw new Error('failed to get full object');
```

Comment:
Yeah, it's ideal if we can just send the error back. We have to come up with a cleaner implementation where we can adjust the headers accordingly.